modules/carousel/jetpack-carousel.js

```diff
@@ -975,6 +981,27 @@ jQuery(document).ready(function($) {
 
 			return args.orig_file;
 		},
+		
+		getImageSizeParts: function(file, orig_width, isPhotonUrl) {
+			var size		= isPhotonUrl ? 
+							file.replace(/.*=([\d]+%2C[\d]+).*$/, '$1') : 
+							file.replace(/.*-([\d]+x[\d]+)\..+$/, '$1');
+						
+			var size_parts  = (size !== file) ? 
+							(isPhotonUrl ? size.split('%2C') : size.split('x')) :
+							[orig_width, 0];
+							
+			if (size_parts[0] === '9999')
+			{
+				size_parts[0] = '0';
+			}
+			if (size_parts[1] === '9999')
+			{
+				size_parts[1] = '0';
+			}
+			
```

Comment:
As far as I'm aware, the dimension photon uses are those in the media settings. In the media settings the 9999 value means that the resize should only consider the other dimension. So if you set the Large file size to 640x9999 and no cropping, it means the image will be resized to 640 and the height will become whatever is needed to maintain the aspect ratio. But for the carousel, the code would interpret 9999 as the actual value of the height, which would produce incorrect results. The actual value is unknown, so that's why I just set it to 0. That way, only the other dimension will determine which image size is used by Photon in the carousel. It's not ideal, but it's the easiest fix I could think of without taking a completely different approach to the original code.